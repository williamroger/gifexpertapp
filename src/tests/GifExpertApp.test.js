import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Tests form GifExpertApp', () => {
  test('should be render GifExpertApp', () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should be render GifExpertApp with two components GifGrid', () => {
    const categories = ['Star Wars', 'Spiderman'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});