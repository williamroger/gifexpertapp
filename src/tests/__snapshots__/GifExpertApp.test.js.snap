// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests form GifExpertApp should be render GifExpertApp 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Tests form GifExpertApp should be render GifExpertApp with two components GifGrid 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Star Wars"
      key="Star Wars"
    />
    <GifGrid
      category="Spiderman"
      key="Spiderman"
    />
  </ol>
</Fragment>
`;
